implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -133,40 +133,63 @@ private DimensionList parseDefaultDimensions(Map<String, String> defaultDimensio
      */
     @Override
     public void export(List<Meter> meters) {
+        Map<String, String> seenMetadata = null;
+        if (config.exportMeterMetadata()) {
+            seenMetadata = new HashMap<>();
+        }
+
         int partitionSize = Math.min(config.batchSize(), DynatraceMetricApiConstants.getPayloadLinesLimit());
         List<String> batch = new ArrayList<>(partitionSize);
 
         for (Meter meter : meters) {
             // Lines that are too long to be ingested into Dynatrace, as well as lines
             // that contain NaN or Inf values are not returned from "toMetricLines",
             // and are therefore dropped.
-            Stream<String> metricLines = toMetricLines(meter);
+            Stream<String> metricLines = toMetricLines(meter, seenMetadata);
 
             metricLines.forEach(line -> {
                 batch.add(line);
-                if (batch.size() == partitionSize) {
-                    send(batch);
-                    batch.clear();
+                trySendBatch(batch, partitionSize);
+            });
+        }
+
+        // if the config to export metadata is turned off, the seenMetadata map will be
+        // null.
+        if (seenMetadata != null) {
+            seenMetadata.values().forEach(line -> {
+                if (line != null) {
+                    batch.add(line);
+                    trySendBatch(batch, partitionSize);
                 }
             });
         }
 
+        // push remaining lines if any.
         if (!batch.isEmpty()) {
             send(batch);
         }
     }
 
-    private Stream<String> toMetricLines(Meter meter) {
-        return meter.match(this::toGaugeLine, this::toCounterLine, this::toTimerLine, this::toDistributionSummaryLine,
-                this::toLongTaskTimerLine, this::toTimeGaugeLine, this::toFunctionCounterLine,
-                this::toFunctionTimerLine, this::toMeterLine);
+    private void trySendBatch(List<String> batch, int partitionSize) {
```

Comment:
The `try` wording had me think of an operation that's likely to fail or put in a try/catch block, which is not the case here.
Maybe `sendBatchIfFull` or `sendBatchIfPartitionFull` might be a clearer name but either should work.